### PR TITLE
api/auth: Replace ioutil with io/os and cleanup

### PR DESCRIPTION
**Pull request by sashamelentyev**

### CODE REVIEW

File: api/auth/azure/azure.go [hunk 1]
```diff
@@ -172,23 +172,21 @@ func (a *AzureAuth) getJWT() (string, error) {
...
 		return "", fmt.Errorf("error reading response body from Azure token endpoint: %w", err)
 	}
 
 	if resp.StatusCode != http.StatusOK {
 		var errResp errorJSON
-		err = json.Unmarshal(responseBytes, &errResp)
-		if err != nil {
-			return "", fmt.Errorf("received error message but was unable to unmarshal its contents")
+		if err := json.Unmarshal(responseBytes, &errResp); err != nil {
+			return "", errors.New("received error message but was unable to unmarshal its contents")
```
Comment (hghaf099) [inline api/auth/azure/azure.go:183]: Although there are many places that we have returned `errors.New`, we would like to keep the `fmt.Errorf` for returning error from functions.

File: api/auth/azure/azure.go [hunk 2]
```diff
@@ -219,23 +217,21 @@ func getMetadata() (metadataJSON, error) {
...
 		return metadataJSON{}, fmt.Errorf("error reading response body from metadata endpoint: %w", err)
 	}
 
 	if resp.StatusCode != http.StatusOK {
 		var errResp errorJSON
-		_ = json.Unmarshal(responseBytes, &errResp)
-		if err != nil {
-			return metadataJSON{}, fmt.Errorf("received error message but was unable to unmarshal its contents")
+		if err := json.Unmarshal(responseBytes, &errResp); err != nil {
+			return metadataJSON{}, errors.New("received error message but was unable to unmarshal its contents")
```
Comment (hghaf099) [inline api/auth/azure/azure.go:228]: Similarly here please.

**Review by hghaf099 — COMMENTED** [review]
Thanks for your contribution. I just posted a couple of minor comments. Otherwise looks good to me.

File: api/auth/azure/azure.go (hunk 1 above)
Comment (sashamelentyev) [reply api/auth/azure/azure.go]: Done

File: api/auth/azure/azure.go (hunk 2 above)
Comment (sashamelentyev) [reply api/auth/azure/azure.go]: Done

**Pull request closed without merging**